packages/webhook-utils/src/update-webhooks.ts

```diff
@@ -0,0 +1,166 @@
+import { WebhookManifest } from "@saleor/app-sdk/types";
+import { Client } from "urql";
+import { webhooksToRemove } from "./filters/webhooks-to-remove";
+import { getWebhookIdsAndManifestsToUpdate } from "./filters/get-webhook-ids-and-manifests-to-update";
+import { webhooksToAdd } from "./filters/webhooks-to-add";
+import { createLogger } from "@saleor/apps-shared";
+import { removeAppWebhook } from "./operations/remove-app-webhook";
+import { WebhookDetailsFragment } from "../generated/graphql";
+import { createAppWebhookFromManifest } from "./create-app-webhook-from-manifest";
+import { modifyAppWebhookFromManifest } from "./modify-app-webhook-from-manifest";
+import { createAppWebhookFromWebhookDetailsFragment } from "./create-app-webhook-from-webhook-details-fragment";
+import { modifyAppWebhookFromWebhookDetails } from "./modify-app-webhook-from-webhook-details";
+
+const logger = createLogger({ name: "updateWebhooks" });
+
+interface RollbackArgs {
+  client: Client;
+  webhookManifests: Array<WebhookManifest>;
+  existingWebhooksData: Array<WebhookDetailsFragment>;
+  addedWebhooks: Array<WebhookDetailsFragment>;
+  modifiedWebhooks: Array<WebhookDetailsFragment>;
+  removedWebhooks: Array<WebhookDetailsFragment>;
+}
+
+const rollback = async ({
+  client,
+  addedWebhooks,
+  modifiedWebhooks,
+  existingWebhooksData,
+  removedWebhooks,
+}: RollbackArgs) => {
+  if (addedWebhooks.length) {
+    logger.info("Removing added webhooks");
+    await Promise.all(
+      addedWebhooks.map((webhook) => removeAppWebhook({ client, webhookId: webhook.id })),
+    );
+  }
+
+  if (modifiedWebhooks.length) {
+    logger.info("Rollback modified webhooks");
+    await Promise.all(
```

Comment:
Wouldn't allSettled be better here?
```suggestion
    await Promise.allSettled(
```

Reply:
Seems like a good suggestion, changed